src/test/java/integ/publish_to_local_queue/PublishToLocalQueueProcess.java

```diff
@@ -0,0 +1,141 @@
+
+import com.google.common.collect.ImmutableList;
+import io.xdb.core.communication.Communication;
+import io.xdb.core.encoder.JacksonJsonObjectEncoder;
+import io.xdb.core.process.Process;
+import io.xdb.core.state.AsyncState;
+import io.xdb.core.state.StateDecision;
+import io.xdb.core.state.StateSchema;
+import io.xdb.gen.models.AsyncStateExecuteRequest;
+import io.xdb.gen.models.CommandRequest;
+import io.xdb.gen.models.CommandWaitingType;
+import io.xdb.gen.models.LocalQueueCommand;
+import io.xdb.gen.models.LocalQueueMessage;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PublishToLocalQueueProcess implements Process {
+
+    @Override
+    public StateSchema getStateSchema() {
+        return StateSchema.withStartingState(new PublishToLocalQueueStartingState(), new PublishToLocalQueueState1());
+    }
+}
+
+class PublishToLocalQueueStartingState implements AsyncState<Void> {
+
+    @Override
+    public Class<Void> getInputType() {
+        return Void.class;
+    }
+
+    @Override
+    public CommandRequest waitUntil(final Void input, final Communication communication) {
+        System.out.println("PublishToLocalQueueStartingState.waitUntil: " + input);
+
+        // will be consumed by PublishToLocalQueueState1
+        communication.publishToLocalQueue(QUEUE_2, PAYLOAD_2);
+
+        return new CommandRequest()
+            .waitingType(CommandWaitingType.ANYOFCOMPLETION)
+            .localQueueCommands(
+                ImmutableList.of(new LocalQueueCommand().queueName(QUEUE_1), new LocalQueueCommand().queueName(QUEUE_3))
+            );
+    }
+
+    @Override
+    public StateDecision execute(
+        final Void input,
+        final Communication communication,
+        final AsyncStateExecuteRequest request
+    ) {
+        System.out.println("PublishToLocalQueueStartingState.execute: " + input);
+
+        // will be consumed by PublishToLocalQueueState1
+        communication.publishToLocalQueue(QUEUE_1, DEDUP_ID, PAYLOAD_1);
+
+        final List<LocalQueueMessage> localQueueResults = request.getCommandResults().getLocalQueueResults();
+        Assertions.assertEquals(1, localQueueResults.size());
+        Assertions.assertEquals(QUEUE_1, localQueueResults.get(0).getQueueName());
+        Assertions.assertEquals(
+            PAYLOAD_1_2,
+            new JacksonJsonObjectEncoder().decode(localQueueResults.get(0).getPayload(), String.class)
+        );
+
+        return StateDecision.singleNextState(PublishToLocalQueueState1.class, null);
+    }
+}
+
+class PublishToLocalQueueState1 implements AsyncState<Void> {
+
+    @Override
+    public Class<Void> getInputType() {
+        return Void.class;
+    }
+
+    @Override
+    public CommandRequest waitUntil(final Void input, final Communication communication) {
+        System.out.println("PublishToLocalQueueState1.waitUntil: " + input);
+
+        // will be consumed by itself
```

Comment:
I actually throwed an error for this case in iWF because it doesn't make too much sense to do so -- it may indicate a mistake in the user process code